src/reducers/versions.spec.tsx

```diff
@@ -1410,16 +1474,46 @@ describe(__filename, () => {
     });
 
     it('dispatches abortFetchDiff() when API call has failed', async () => {
+      const addonId = 1;
+      const baseVersionId = 2;
+      const headVersionId = 2;
       const _getDiff = jest.fn().mockReturnValue(
         Promise.resolve({
           error: new Error('Bad Request'),
         }),
       );
 
-      const { dispatch, thunk } = _fetchDiff({ _getDiff });
+      const { dispatch, thunk } = _fetchDiff({
+        _getDiff,
+        addonId,
+        baseVersionId,
+        headVersionId,
+      });
       await thunk();
 
-      expect(dispatch).toHaveBeenCalledWith(actions.abortFetchDiff());
+      expect(dispatch).toHaveBeenCalledWith(
+        actions.abortFetchDiff({
+          addonId,
+          baseVersionId,
+          headVersionId,
+        }),
+      );
+    });
+
+    it('prevents itself to execute more than once for the same diff', async () => {
+      const addonId = 1;
+      const baseVersionId = 2;
+      const headVersionId = 3;
+
+      const { dispatch, thunk, store } = _fetchDiff();
```

Comment:
This also needs to pass in `addonId`, `baseVersionId`, and `headVersionId` so it's not relying on the defaults.